### PR TITLE
Don’t parse unicode escapes

### PR DESCRIPTION
**Pull request by mbostock**
A consequence of this is that the `cooked` value is exclusively null, but I think that makes sense, since we only have a raw value if we don’t parse unicode escape sequences.

### CODE REVIEW

File: tap-snapshots/test-parse-test.js-TAP.test.js [hunk 1]
```diff
@@ -675,16 +675,16 @@ exports[`test/parse-test.js TAP parse backslash-markdown.md > must match snapsho
 Node {
   "async": false,
   "body": Node {
-    "end": 27,
+    "end": 28,
```
Comment (mbostock) [inline tap-snapshots/test-parse-test.js-TAP.test.js:678]: Can you check if this 27 ↦ 28 is correct, @duaneatat? I wonder if there’s an edge case in the new readTemplateToken that I’m missing here.

Comment (duaneatat) [reply tap-snapshots/test-parse-test.js-TAP.test.js:678]: will do!

Comment (duaneatat) [reply tap-snapshots/test-parse-test.js-TAP.test.js:678]: it looks like it's off by one if there is a trailing `\` -- trying to figure out why

Comment (duaneatat) [reply tap-snapshots/test-parse-test.js-TAP.test.js:678]: maybe we need to call `finishToken` in the `\\` case if there isn't another character coming next?

File: src/parse.js [hunk 2]
```diff
@@ -341,50 +341,24 @@ export class TemplateCellParser extends CellParser {
...
-      case 13:
-        if (this.input.charCodeAt(this.pos) === 10) ++this.pos; // falls through
-      case 10:
-        out += "\n";
-        break;
-      default:
-        out += String.fromCharCode(ch);
+  out: for (; this.pos < this.input.length; this.pos++) {
+    switch (this.input[this.pos]) {
+      case "\\": ++this.pos; break;
```
Comment (duaneatat) [inline src/parse.js:348]: maybe we need to call `finishToken` in this case if there isn't another character coming next and not increment?

Comment (duaneatat) [reply src/parse.js:348]: was comparing this to https://github.com/acornjs/acorn/blob/556a7ab62dd999c9024ed0e0920ebb5f6521c4de/acorn/src/tokenize.js#L672-L692

but that doesn't seem to be necessary since the boundaries of the `for` loop already enforce that

Comment (mbostock) [reply src/parse.js:348]: I will tack on a fix for the out of bounds with trailing slash. Thanks for tracking it down!

File: src/parse.js [hunk 3]
```diff
@@ -340,51 +340,28 @@ export class TemplateCellParser extends CellParser {
...
-        if (this.input.charCodeAt(this.pos) === 10) ++this.pos; // falls through
-      case 10:
-        out += "\n";
-        break;
-      default:
-        out += String.fromCharCode(ch);
+  out: for (; this.pos < this.input.length; this.pos++) {
+    switch (this.input.charCodeAt(this.pos)) {
+      case 92: { // slash
+        if (this.pos < this.input.length - 1) ++this.pos; // terminal slash
```
Comment (duaneatat) [inline src/parse.js:349]:
```suggestion
        if (this.pos < this.input.length - 1) ++this.pos; // if not a terminal slash
```

**Review by duaneatat — APPROVED** [review]
🎯 lgtm!
